pkgs/by-name/gi/git-pr/package.nix

```diff
@@ -0,0 +1,27 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+
+buildGoModule rec {
+  pname = "git-pr";
+  version = "0.0.2";
+
+  src = fetchFromGitHub {
+    owner = "picosh";
+    repo = "git-pr";
+    rev = "v${version}";
+    hash = "sha256-7Ka8p5X8nQBXKiT6QsWOWMQJL8rePKrHz/LZU1W+oQ8=";
+  };
+
+  vendorHash = "sha256-tu5C7hz6UTgn/jCCotXzZHlUmGVNERhA7Osxi31Domk=";
+
+  meta = {
+    homepage = "https://pr.pico.sh";
+    description = "Simple git collaboration tool";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ sigmanificient ];
+    mainProgram = "ssh";
```

Comment:
This could conflict with the actual ssh?

Reply:
No i dont think so, mainProgram is use to designate of one the binaries provided by the derivation as the main one

Reply:
but if added to the same environment as openssh they will conflict in PATH and any symlinktrees containing all binaries. Please rename the binary

Reply:
I wonder why it provides the "ssh" binary at all.

Reply:
what should I rename it to?

Reply:
Just

```nix
postInstall = ''
  mv $out/bin/ssh $out/bin/git-pr
'';
```

Reply:
> I wonder why it provides the "ssh" binary at all.

I guess this is intended. https://github.com/picosh/git-pr?tab=readme-ov-file#format-patch-workflow

It provides a custom ssh binary for the workflow. It will conflict and most likely won't be usable with `openssh` and this even may break compatibility with the openssh server module we have in NixOS.